packaging/rpm/rear.spec

```diff
@@ -18,8 +18,7 @@ License: GPL-3.0
 Group: Applications/File
 URL: http://relax-and-recover.org/
 
-# as GitHub stopped with download section we need to go back to Sourceforge for downloads
-Source: https://sourceforge.net/projects/rear/files/rear/%{version}/rear-%{version}.tar.gz
+Source: https://github.com/rear/rear/archive/%{version}.tar.gz#/rear-%{version}.tar.gz
```

Comment:
Nice, didn't know this trick. Do we need the `#...` suffix? I tried several URLs like

* https://github.com/rear/rear/archive/galaxy11.tar.gz
* https://github.com/rear/rear/archive/531d9074d584ba4192da51ee5b7ab0fad460d57b.tar.gz
* https://github.com/rear/rear/archive/2.7.tar.gz
* https://github.com/rear/rear/archive/master.tar.gz

And they all yielded archives with a matching `rear-<tag/branch/commit>` names *and* subdirectories

Reply:
This trick is documented here: https://docs.fedoraproject.org/en-US/packaging-guidelines/SourceURL/#_troublesome_urls , with examples here: https://docs.fedoraproject.org/en-US/packaging-guidelines/SourceURL/#_git_tags .
In general, it is needed so that the tarballs are named `%{project}-%{version}.tar.gz` and not just `%{version}.tar.gz` in the case of a typical archive URL like `https://github.com/rear/rear/archive/%{version}.tar.gz`.
I don't think you get a `rear-<tag/branch/commit>`  name with such an URL, only `<tag/branch/commit>` (`.tar.gz`). See:
```
wget https://github.com/rear/rear/archive/2.7.tar.gz
--2023-03-01 12:47:12--  https://github.com/rear/rear/archive/2.7.tar.gz
Resolving github.com (github.com)... 140.82.114.3
Connecting to github.com (github.com)|140.82.114.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/rear/rear/tar.gz/refs/tags/2.7 [following]
--2023-03-01 12:47:13--  https://codeload.github.com/rear/rear/tar.gz/refs/tags/2.7
2023-03-01 12:47:14 (1.45 MB/s) - ‘2.7.tar.gz’ saved [852253]
```
And note that this URL is not used just to know where to download the file from. `rpmbuild` extracts its last component (basename) to know what is the local archive name, even if the file is pre-downloaded. `2.7.tar.gz` is not correct.
In the case of ReaR, the trick is unnecessary, because there have always been `rear-%{version}` tags in addition to the bare `%{version}` tags in the repo (this is not typical for most projects/repositories). Thanks to this, I can simply specify `Source:` as `https://github.com/rear/rear/archive/%{name}-%{version}.tar.gz` and get rid of the trick.

Reply:
Ah, I learned something new: `curl` and `wget` ignore - by default - the `content-disposition: attachment; filename=rear-2.7.tar.gz` header that GitHub actually sets.

`curl -OJ` actually works:
<img width="849" alt="image" src="https://user-images.githubusercontent.com/101384/222220220-48592ad9-4689-4092-96ad-a93323f09736.png">

Good that you found a way around that problem

Reply:
Ah, I learned something new as well, I did not know that there is such a header (I assumed that wget would use it if it were set).
Anyway, respecting it would not have helped, because the file would get downloaded under the correct name, but rpmbuild would not be able to figure out what the name is (as it extracts the last component of the URL to get the file name).

Reply:
> In the case of ReaR, the trick is unnecessary, because there have always been `rear-%{version}` tags in addition to the bare `%{version}` tags in the repo (this is not typical for most projects/repositories). Thanks to this, I can simply specify `Source:` as `https://github.com/rear/rear/archive/%{name}-%{version}.tar.gz` and get rid of the trick.

It turns out that I can not, because https://github.com/rear/rear/archive/rear-2.7.tar.gz is a text file that says only
```
the given path has multiple possibilities: #<Git::Ref:0x00007f4e90a15960>, #<Git::Ref:0x00007f4e90a14f60>
```
This is probably because there is both a branch named [rear-2.7](https://github.com/rear/rear/tree/rear-2.7) and a tag named [rear-2.7](https://github.com/rear/rear/releases/tag/rear-2.7), and similarly for the older releases (2.6, 2.5). So I have to go back to the original idea of using bare `%{version}`, in the archive name and keep the trick, like 2.7.tar.gz.

In the future it would probably be a good idea to use a release branch name that does not shadow the release tag name, like e.g. `release-2.8`.

Reply:
@pcahyna 
I did the ReaR 2.8 release (my first release ever) according to
https://github.com/rear/rear/wiki/Release-process
as far as I could understand the instructions there and
partially in some kind of "bona fide script kiddie" mode ;-)
so feel free to enhance
https://github.com/rear/rear/wiki/Release-process
as needed.